app/models/perm.rb

```diff
@@ -9,18 +9,38 @@
 #
 
 class Perm < ActiveRecord::Base
-  ##
-  # Associations
-  has_and_belongs_to_many :users, :join_table => :users_perms
-
-  validates :name, presence: {message: _("can't be blank")}, uniqueness: {message: _("must be unique")}
-
-  scope :add_orgs, -> {Perm.find_by(name: 'add_organisations')}
-  scope :change_affiliation, -> {Perm.find_by(name: 'change_org_affiliation')}
-  scope :grant_permissions, -> {Perm.find_by(name: 'grant_permissions')}
-  scope :modify_templates, -> {Perm.find_by(name: 'modify_templates')}
-  scope :modify_guidance, -> {Perm.find_by(name: 'modify_guidance')}
-  scope :use_api, -> {Perm.find_by(name: 'use_api')}
-  scope :change_org_details, -> {Perm.find_by(name: 'change_org_details')}
-  scope :grant_api, -> {Perm.find_by(name: 'grant_api_to_orgs')}
+  include ValidationMessages
+
+  # ================
+  # = Associations =
+  # ================
+
+  has_and_belongs_to_many :users, join_table: :users_perms
+
+  # ===============
+  # = Validations =
+  # ===============
+
+  validates :name, presence: { message: PRESENCE_MESSAGE },
+                   uniqueness: { message: UNIQUENESS_MESSAGE }
+
+  # ==========
+  # = Scopes =
+  # ==========
+
+  scope :add_orgs, -> { Perm.find_by(name: 'add_organisations') }
+
+  scope :change_affiliation, -> { Perm.find_by(name: 'change_org_affiliation') }
+
+  scope :grant_permissions, -> { Perm.find_by(name: 'grant_permissions') }
+
+  scope :modify_templates, -> { Perm.find_by(name: 'modify_templates') }
+
+  scope :modify_guidance, -> { Perm.find_by(name: 'modify_guidance') }
+
+  scope :use_api, -> { Perm.find_by(name: 'use_api') }
+
+  scope :change_org_details, -> { Perm.find_by(name: 'change_org_details') }
+
+  scope :grant_api, -> { Perm.find_by(name: 'grant_api_to_orgs') }
```

Comment:
Seeing as these records never change, would there be a downside to making these constants?
It would likely save quite a few queries, but these are so frequently accessed they likely get cached

Reply:
They won't be cached across requests.

Instead of making them constants, it might be a better approach to explicitly cache them. E.g.

``` ruby
def self.grant_api
  cache.fetch(["perm", "grant_api"], expires_in: 1.hour) { Perm.find_by(name: 'grant_api_to_orgs') }
end
```

This would require some caching store to be set up. Most likely Redis or Memcached.